userspace/libsinsp/event.cpp

```diff
@@ -2585,96 +2579,68 @@ const sinsp_evt_param* sinsp_evt::get_param_value_raw(const char* name)
 
 void sinsp_evt::get_category(OUT sinsp_evt::category* cat)
 {
-	if(m_pevt->type == PPME_GENERIC_E ||
-		m_pevt->type == PPME_GENERIC_X)
-	{
-		//
-		// This event is a syscall that doesn't have a filler yet.
-		// The category can be found in g_syscall_info_table.
-		//
-		sinsp_evt_param *parinfo = get_param(0);
-		ASSERT(parinfo->m_len == sizeof(uint16_t));
-		uint16_t id = *(uint16_t *)parinfo->m_val;
+	/* We always search the category inside the event table */
+	cat->m_category = get_category();
 
-		if(id < PPM_SC_MAX)
+	//
+	// For EC_IO and EC_WAIT events, we dig into the fd state to get the category
+	// and fdtype
+	//
+	if(cat->m_category & EC_IO_BASE)
```

Comment:
Could we double check all these cases, because that approach really did not work for me, returned no valid results, see https://github.com/falcosecurity/libs/pull/835.
@Andreagit97 and @FedeDP , ty!